custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -784,57 +801,45 @@ def _prepare_prediction_server_or_batch_pipeline(self, run_language):
 
         return DrumUtils.render_file(functional_pipeline_filepath, replace_data)
 
-    def _run_predictions(self):
-        if self.run_mode != RunMode.SCORE:
-            raise NotImplemented("Only score mode is supported here")
+    def _run_predictions(self, stats_collector: Optional[StatsCollector] = None):
+        if self.run_mode not in [RunMode.SCORE, RunMode.SERVER]:
+            raise NotImplemented(f"The given run mode is supported here: {self.run_mode}")
 
-        with self._setup_output_if_not_exists():
-            run_language = self._check_artifacts_and_get_run_language()
-            infra_pipeline_str = self._prepare_prediction_server_or_batch_pipeline(run_language)
-
-            pipeline = json.loads(infra_pipeline_str)
-            if "pipe" not in pipeline or not pipeline["pipe"]:
-                raise DrumCommonException("Pipeline is empty")
-            if "arguments" not in pipeline["pipe"][0]:
-                raise DrumCommonException("Arguments are missing in the pipeline")
-
-            self.logger.info(
-                f">>> Start {ArgumentsOptions.MAIN_COMMAND} in the {self.run_mode.value} mode"
-            )
+        run_language = self._check_artifacts_and_get_run_language()
+        infra_pipeline_str = self._prepare_prediction_server_or_batch_pipeline(run_language)
 
-            sc = StatsCollector(
-                disable_instance=(
-                    not hasattr(self.options, "show_perf")
-                    or not self.options.show_perf
-                    or self.run_mode == RunMode.SERVER
-                )
-            )
-            sc.register_report("Full time", "end", StatsOperation.SUB, "start")
-            sc.register_report(
-                "Init time (incl model loading)", "init", StatsOperation.SUB, "start"
-            )
-            sc.register_report("Run time (incl reading CSV)", "run", StatsOperation.SUB, "init")
-            sc.enable()
+        pipeline = json.loads(infra_pipeline_str)
+        if "pipe" not in pipeline or not pipeline["pipe"]:
+            raise DrumCommonException("Pipeline is empty")
+        if "arguments" not in pipeline["pipe"][0]:
+            raise DrumCommonException("Arguments are missing in the pipeline")
 
-            try:
-                sc.mark("start")
-                predictor = GenericPredictorComponent()
-                predictor.configure(pipeline["pipe"][0]["arguments"])
-                sc.mark("init")
+        self.logger.info(
+            f">>> Start {ArgumentsOptions.MAIN_COMMAND} in the {self.run_mode.value} mode"
+        )
 
-                predictor.materialize()
-                sc.mark("run")
-            finally:
-                predictor.terminate()
-                sc.mark("end")
-                sc.disable()
+        predictor = (
+            PredictionServer() if self.run_mode == RunMode.SERVER else GenericPredictorComponent()
+        )
+        try:
+            if stats_collector:
```

Comment:
I am nitpicking here: have you considered splitting this `_run_predictions` into two separate functions for RunMode.SCORE  & RunMode.SERVER just to omit these if statements?